queue/abstract/driver/utube.lua

```diff
@@ -98,12 +123,61 @@ function method.put(self, data, opts)
 
     local id = max and max[1] + 1 or 0
     local task = self.space:insert{id, state.READY, tostring(opts.utube), data}
+    if self.v2 then
+        put_ready(self, id, task[3])
+    end
     self.on_task_change(task, 'put')
     return task
 end
 
+local function take_ready(self)
+    for s, task_ready in self.space_ready:pairs({}, { iterator = 'GE' }) do
```

Comment:
```suggestion
    for s, task_ready in self.space_ready:pairs({}, { iterator = 'GE' }) do
```
It is a little confusing here: we don't need to iterate by the index, by try to peek up a first item and repeat if no success.

Reply:
But what if the first element is in `READY` state but the utube it is in has `TAKEN` state? Then we will be waiting for utube to leave the `TAKEN` state first, instead of trying the element from the next utube.

Reply:
We need just to repeat to try to take a first task from the space.

The task in the `_ready` becomes only if the `utube` have not a tasks in the `TAKEN` state.
But we could see a task in the `_ready` space if the `utube` have tasks in the `TAKEN` state due to concurrent updated. But we expect that it gone soon. Maybe we need a `yield` between tries.